### PR TITLE
Adds GET /signups Rogue Proxy

### PR DESCRIPTION
**Pull request by katiecrane**
#### What's this PR do?
Example responses are below. There are several parts to these responses, and a few of them are constructed the exact same way that they are in the original Phoenix response. 
Parts of a response:
- Meta data
    - This is transformed from the Rogue response
- Signup
    - We get the signup data from Rogue and transform it
- Campaign
    - this is created the same way that Phoenix does it, using [`transformCampaign`](https://github.com/DoSomething/phoenix/blob/dev/lib/modules/dosomething/dosomething_api/includes/Transformer.php#L191-L274)
- User
    - this is created the same way that Phoenix does it, using [`transformUser`](https://github.com/DoSomething/phoenix/blob/dev/lib/modules/dosomething/dosomething_api/includes/Transformer.php#L459-L468)
    - note that we pass the Northstar user when Rogue is on
- Reportback
    - We get the reportback data from Rogue and transform it

Please pay special attention to:
Everything about runs. [This line in particular.](https://github.com/DoSomething/phoenix/pull/7432/files#diff-75aae572ad7d94de41778845b7683db0R147) My concern for this line is I'm not sure that it necessarily gets the same data as [this call in the Phoenix SignupTransformer](https://github.com/DoSomething/phoenix/blob/dev/lib/modules/dosomething/dosomething_signup/includes/SignupTransformer.php#L132).

**Example Response with Rogue on**
I commented up the first element of this response
Request `http://dev.dosomething.org:8888/api/v1/signups?count=3`
```
{
    "meta": {
        "pagination": {
            "total": 4596,
            "per_page": 3,
            "current_page": 1,
            "total_pages": 1532,
            "links": {
                "prev_uri": null,
                "next_uri": "http://dev.dosomething.org:8888/api/v1/signups?count=3&page=2"
            }
        }
    },
    "data": [
        {
            "id": 9516,
            "created_at": 1457977530,
            "campaign_run": {
                "id": "1862",
                "current": false,
                "timing": {
                    "start": "2014-10-02 00:00:00",
                    "end": null
                }
            },
            "uri": null,
            "campaign": {
                "id": "1503",
                "title": "Musical Mixup",
                "campaign_runs": null,
                "language": "en",
                "translations": {
                    "original": "en",
                    "data": {
                        "en": {
                            "entity_type": "node",
                            "entity_id": "1503",
                            "revision_id": "4694",
                            "language": "en",
                            "source": "",
                            "uid": "4",
                            "status": "1",
                            "translate": "0",
                            "created": "1412258729",
                            "changed": "1451949650"
                        }
                    }
                },
                "tagline": null,
                "status": "1",
                "type": "campaign"
            },
            "user": {
                "id": "559442cfa59dbfc9578b4c14",
                "drupal_id": "10",
                "first_name": "Andrea",
                "last_initial": "G",
                "photo": null,
                "country": "US"
            },
            "reportback": {
                "id": 9516,
                "created_at": 1457977530,
                "updated_at": 1458762952,
                "quantity": 21,
                "uri": null,
                "why_participated": "asdf",
                "flagged": null,
                "reportback_items": {
                    "total": 1,
                    "data": [
                        {
                            "id": 1923,
                            "status": "approved",
                            "caption": "arg",
                            "uri": null,
                            "media": {
                                "uri": "default",
                                "type": "image"
                            },
                            "created_at": 1458762952,
                            "kudos": {
                                "data": {
                                    "641": {
                                        "term": {
                                            "id": "641",
                                            "name": "heart",
                                            "total": 1
                                        },
                                        "current_user": null
                                    }
                                }
                            }
                        }
                    ]
                }
            }
        },
        {
            "id": 9517,
            "created_at": 1457977840,
            "campaign_run": {
                "id": "1903",
                "current": false,
                "timing": {
                    "start": "2016-02-07 00:00:00",
                    "end": "2018-12-31 00:00:00"
                }
            },
            "uri": null,
            "campaign": {
                "id": "1631",
                "title": "Space Jam",
                "campaign_runs": null,
                "language": "en",
                "translations": {
                    "original": "en",
                    "data": {
                        "en": {
                            "entity_type": "node",
                            "entity_id": "1631",
                            "revision_id": "5082",
                            "language": "en",
                            "source": "",
                            "uid": "8",
                            "status": "1",
                            "translate": "0",
                            "created": "1426628406",
                            "changed": "1470409442"
                        },
                        "en-global": {
                            "entity_type": "node",
                            "entity_id": "1631",
                            "revision_id": "5082",
                            "language": "en-global",
                            "source": "en",
                            "uid": "9",
                            "status": "0",
                            "translate": "0",
                            "created": "1454892338",
                            "changed": "1454892338"
                        },
                        "pt-br": {
                            "entity_type": "node",
                            "entity_id": "1631",
                            "revision_id": "5082",
                            "language": "pt-br",
                            "source": "en",
                            "uid": "10",
                            "status": "0",
                            "translate": "0",
                            "created": "1445625044",
                            "changed": "1454871600"
                        }
                    }
                },
                "tagline": null,
                "status": "1",
                "type": "campaign"
            },
            "user": {
                "id": "56e6f9f1a59dbf63358b45a9",
                "drupal_id": "1706193",
                "first_name": "Mina",
                "last_initial": "",
                "photo": null,
                "country": "US"
            },
            "reportback": {
                "id": 9517,
                "created_at": 1457977840,
                "updated_at": 1457977840,
                "quantity": null,
                "uri": null,
                "why_participated": null,
                "flagged": null,
                "reportback_items": {
                    "total": 0,
                    "data": []
                }
            }
        },
        {
            "id": 9518,
            "created_at": 1457984853,
            "campaign_run": {
                "id": "1862",
                "current": false,
                "timing": {
                    "start": "2014-10-02 00:00:00",
                    "end": null
                }
            },
            "uri": null,
            "campaign": {
                "id": "1503",
                "title": "Musical Mixup",
                "campaign_runs": null,
                "language": "en",
                "translations": {
                    "original": "en",
                    "data": {
                        "en": {
                            "entity_type": "node",
                            "entity_id": "1503",
                            "revision_id": "4694",
                            "language": "en",
                            "source": "",
                            "uid": "4",
                            "status": "1",
                            "translate": "0",
                            "created": "1412258729",
                            "changed": "1451949650"
                        }
                    }
                },
                "tagline": null,
                "status": "1",
                "type": "campaign"
            },
            "user": {
                "id": "56e71557a59dbfe3448b4580",
                "drupal_id": "1706195",
                "first_name": "N",
                "last_initial": "",
                "photo": null,
                "country": "US"
            },
            "reportback": {
                "id": 9518,
                "created_at": 1457984853,
                "updated_at": 1457984853,
                "quantity": null,
                "uri": null,
                "why_participated": null,
                "flagged": null,
                "reportback_items": {
                    "total": 0,
                    "data": []
                }
            }
        }
    ]
}
```

**Response from same request with Rogue OFF**
```
{
    "meta": {
        "pagination": {
            "total": 4800,
            "per_page": 3,
            "current_page": 1,
            "total_pages": 1600,
            "links": {
                "prev_uri": null,
                "next_uri": "http://dev.dosomething.org:8888/api/v1/signups?count=3&page=2"
            }
        }
    },
    "data": [
        {
            "id": "4809",
            "created_at": "1496176736",
            "campaign_run": {
                "id": "1837",
                "current": true,
                "timing": {
                    "start": "2014-07-24 00:00:00",
                    "end": null
                }
            },
            "uri": "http://dev.dosomething.org:8888/api/v1/signups/4809",
            "campaign": {
                "id": "1321",
                "title": "Hunt Day 5: Band-Aid Brigade",
                "campaign_runs": {
                    "current": {
                        "en": {
                            "id": "1837"
                        }
                    },
                    "past": []
                },
                "language": {
                    "language_code": "en",
                    "prefix": "us"
                },
                "translations": {
                    "original": "en",
                    "en": {
                        "language_code": "en",
                        "prefix": "us"
                    }
                },
                "tagline": "Send in band-aids to stock a shelter's first-aid supply.",
                "status": "active",
                "type": "campaign",
                "reportback_info": {
                    "copy": "Submit a pic of your band-aid stash to enter for a $10,000 scholarship!",
                    "confirmation_message": "You rocked at collecting band-aids. The shelters will be grateful to get them!",
                    "noun": "band-aids",
                    "verb": "donated"
                },
                "magic_link_copy": null
            },
            "reportback": null,
            "user": {
                "id": "57ab7687a59dbf5a3d8b47ab",
                "drupal_id": "1707996",
                "first_name": "katie",
                "last_initial": "",
                "photo": null,
                "country": "US"
            }
        },
        {
            "id": "4808",
            "created_at": "1496176190",
            "campaign_run": {
                "id": "1816",
                "current": true,
                "timing": {
                    "start": "2014-05-02 00:00:00",
                    "end": null
                }
            },
            "uri": "http://dev.dosomething.org:8888/api/v1/signups/4808",
            "campaign": {
                "id": "1173",
                "title": "Thumb Wars",
                "campaign_runs": {
                    "current": {
                        "en": {
                            "id": "1816"
                        },
                        "en-global": {
                            "id": "1816"
                        }
                    },
                    "past": []
                },
                "language": {
                    "language_code": "en",
                    "prefix": "us"
                },
                "translations": {
                    "original": "en",
                    "en": {
                        "language_code": "en",
                        "prefix": "us"
                    },
                    "en-global": {
                        "language_code": "en-global",
                        "prefix": null
                    }
                },
                "tagline": "Send friends Thumb Socks to remind them not to text and drive.",
                "status": "active",
                "type": "campaign",
                "reportback_info": {
                    "copy": "Send us a shot of your friend rockin' their Socks!",
                    "confirmation_message": "Thumbs up, Thumb Warrior! Thanks for keeping your friends -- and the streets -- safe.",
                    "noun": "Thumb Socks",
                    "verb": "shared"
                },
                "magic_link_copy": null
            },
            "reportback": null,
            "user": {
                "id": "57ab7687a59dbf5a3d8b47ab",
                "drupal_id": "1707996",
                "first_name": "katie",
                "last_initial": "",
                "photo": null,
                "country": "US"
            }
        },
        {
            "id": "4807",
            "created_at": "1496176065",
            "campaign_run": {
                "id": "1833",
                "current": true,
                "timing": {
                    "start": "2016-01-04 00:00:00",
                    "end": null
                }
            },
            "uri": "http://dev.dosomething.org:8888/api/v1/signups/4807",
            "campaign": {
                "id": "1283",
                "title": "Patient Playbooks GL",
                "campaign_runs": {
                    "current": {
                        "en": {
                            "id": "1833"
                        },
                        "en-global": {
                            "id": "1833"
                        },
                        "es-mx": {
                            "id": "1833"
                        },
                        "pt-br": {
                            "id": "1833"
                        }
                    },
                    "past": []
                },
                "language": {
                    "language_code": "en",
                    "prefix": "us"
                },
                "translations": {
                    "original": "en",
                    "en": {
                        "language_code": "en",
                        "prefix": "us"
                    },
                    "en-global": {
                        "language_code": "en-global",
                        "prefix": null
                    },
                    "es-mx": {
                        "language_code": "es-mx",
                        "prefix": "mx"
                    },
                    "pt-br": {
                        "language_code": "pt-br",
                        "prefix": "br"
                    }
                },
                "tagline": "Create activity books for children in hospitals.",
                "status": "active",
                "type": "campaign",
                "reportback_info": {
                    "copy": "Submit a pic of your playbook!",
                    "confirmation_message": "Oh, yeah! Thanks for sending smiles! ",
                    "noun": "Playbooks",
                    "verb": "Sent"
                },
                "magic_link_copy": null
            },
            "reportback": null,
            "user": {
                "id": "57ab7687a59dbf5a3d8b47ab",
                "drupal_id": "1707996",
                "first_name": "katie",
                "last_initial": "",
                "photo": null,
                "country": "US"
            }
        }
    ]
}
```

**Rogue response featuring multiple posts**
`http://dev.dosomething.org:8888/api/v1/signups?user=56f18b9da59dbf59048b456c`
```
{
    "meta": {
        "pagination": {
            "total": 1,
            "per_page": 25,
            "current_page": 1,
            "total_pages": 1,
            "links": {
                "prev_uri": null,
                "next_uri": "http://dev.dosomething.org:8888/api/v1/signups?user=56f18b9da59dbf59048b456c&page=2"
            }
        }
    },
    "data": [
        {
            "id": 9534,
            "created_at": 1458670492,
            "campaign_run": {
                "id": "1903",
                "current": false,
                "timing": {
                    "start": "2016-02-07 00:00:00",
                    "end": "2018-12-31 00:00:00"
                }
            },
            "uri": null,
            "campaign": {
                "id": "1631",
                "title": "Space Jam",
                "campaign_runs": null,
                "language": "en",
                "translations": {
                    "original": "en",
                    "data": {
                        "en": {
                            "entity_type": "node",
                            "entity_id": "1631",
                            "revision_id": "5082",
                            "language": "en",
                            "source": "",
                            "uid": "8",
                            "status": "1",
                            "translate": "0",
                            "created": "1426628406",
                            "changed": "1470409442"
                        },
                        "en-global": {
                            "entity_type": "node",
                            "entity_id": "1631",
                            "revision_id": "5082",
                            "language": "en-global",
                            "source": "en",
                            "uid": "9",
                            "status": "0",
                            "translate": "0",
                            "created": "1454892338",
                            "changed": "1454892338"
                        },
                        "pt-br": {
                            "entity_type": "node",
                            "entity_id": "1631",
                            "revision_id": "5082",
                            "language": "pt-br",
                            "source": "en",
                            "uid": "10",
                            "status": "0",
                            "translate": "0",
                            "created": "1445625044",
                            "changed": "1454871600"
                        }
                    }
                },
                "tagline": null,
                "status": "1",
                "type": "campaign"
            },
            "user": {
                "id": "56f18b9da59dbf59048b456c",
                "drupal_id": "1706281",
                "first_name": "Roberto",
                "last_initial": "G",
                "photo": null,
                "country": "US"
            },
            "reportback": {
                "id": 9534,
                "created_at": 1458670492,
                "updated_at": 1485545565,
                "quantity": 2147483647,
                "uri": null,
                "why_participated": "'''",
                "flagged": null,
                "reportback_items": {
                    "total": 7,
                    "data": [
                        {
                            "id": 1930,
                            "status": "pending",
                            "caption": "Their faces when we won.",
                            "uri": null,
                            "media": {
                                "uri": "default",
                                "type": "image"
                            },
                            "created_at": 1458672335,
                            "kudos": {
                                "data": []
                            }
                        },
                        {
                            "id": 1931,
                            "status": "pending",
                            "caption": "Their faces when we won.",
                            "uri": null,
                            "media": {
                                "uri": "default",
                                "type": "image"
                            },
                            "created_at": 1458672694,
                            "kudos": {
                                "data": []
                            }
                        },
                        {
                            "id": 1932,
                            "status": "pending",
                            "caption": "bunny",
                            "uri": null,
                            "media": {
                                "uri": "default",
                                "type": "image"
                            },
                            "created_at": 1458672945,
                            "kudos": {
                                "data": []
                            }
                        },
                        {
                            "id": 1933,
                            "status": "pending",
                            "caption": "BFG",
                            "uri": null,
                            "media": {
                                "uri": "default",
                                "type": "image"
                            },
                            "created_at": 1458673023,
                            "kudos": {
                                "data": []
                            }
                        },
                        {
                            "id": 1934,
                            "status": "pending",
                            "caption": "duuuude",
                            "uri": null,
                            "media": {
                                "uri": "default",
                                "type": "image"
                            },
                            "created_at": 1458673094,
                            "kudos": {
                                "data": []
                            }
                        },
                        {
                            "id": 1935,
                            "status": "pending",
                            "caption": "silly wabbit",
                            "uri": null,
                            "media": {
                                "uri": "default",
                                "type": "image"
                            },
                            "created_at": 1458686313,
                            "kudos": {
                                "data": []
                            }
                        },
                        {
                            "id": 1936,
                            "status": "pending",
                            "caption": "I'm so small!",
                            "uri": null,
                            "media": {
                                "uri": "default",
                                "type": "image"
                            },
                            "created_at": 1458686433,
                            "kudos": {
                                "data": []
                            }
                        }
                    ]
                }
            }
        }
    ]
}
```

**Phoenix response with multiple posts**
```
{
    "meta": {
        "pagination": {
            "total": 1,
            "per_page": 25,
            "current_page": 1,
            "total_pages": 1,
            "links": {
                "prev_uri": null,
                "next_uri": null
            }
        }
    },
    "data": [
        {
            "id": "3268",
            "created_at": "1458670492",
            "campaign_run": {
                "id": "1903",
                "current": true,
                "timing": {
                    "start": "2016-02-07 00:00:00",
                    "end": "2018-12-31 00:00:00"
                }
            },
            "uri": "http://dev.dosomething.org:8888/api/v1/signups/3268",
            "campaign": {
                "id": "1631",
                "title": "Space Jam",
                "campaign_runs": {
                    "current": {
                        "en": {
                            "id": "1903"
                        },
                        "en-global": {
                            "id": "1903"
                        },
                        "pt-br": {
                            "id": "1874"
                        }
                    },
                    "past": []
                },
                "language": {
                    "language_code": "en",
                    "prefix": "us"
                },
                "translations": {
                    "original": "en",
                    "en": {
                        "language_code": "en",
                        "prefix": "us"
                    },
                    "en-global": {
                        "language_code": "en-global",
                        "prefix": null
                    },
                    "pt-br": {
                        "language_code": "pt-br",
                        "prefix": "br"
                    }
                },
                "tagline": "Here's your chance, do your dance at the Space Jam.   ",
                "status": "active",
                "type": "campaign",
                "reportback_info": {
                    "copy": "Submit a photo of you and the Tune Squad defeating the Monstars.",
                    "confirmation_message": null,
                    "noun": "Looney Tunes",
                    "verb": "Saved"
                },
                "magic_link_copy": "View Space Jam on the web to register to vote!"
            },
            "reportback": {
                "id": "3612",
                "created_at": "2016-03-22T18:45:35+00:00",
                "updated_at": "2017-01-27T19:32:45+00:00",
                "quantity": 2147483647,
                "uri": "http://dev.dosomething.org:8888/api/v1/reportbacks/3612",
                "why_participated": null,
                "flagged": false,
                "reportback_items": {
                    "total": 7,
                    "data": [
                        {
                            "id": "6788",
                            "status": "approved",
                            "caption": "Their faces when we won.",
                            "uri": "http://dev.dosomething.org:8888/api/v1/reportback-items/6788",
                            "media": {
                                "uri": "http://dev.dosomething.org:8888/sites/default/files/styles/480x480/public/reportbacks/1631/uid_1706281-nid_1631-run_nid_1903-0.jpg?itok=2ZEmuWMZ",
                                "type": "image"
                            },
                            "created_at": "1458672335",
                            "kudos": {
                                "data": {
                                    "641": {
                                        "term": {
                                            "id": "641",
                                            "name": "heart",
                                            "total": 2
                                        },
                                        "current_user": null
                                    }
                                }
                            }
                        },
                        {
                            "id": "6789",
                            "status": "approved",
                            "caption": "Their faces when we won.",
                            "uri": "http://dev.dosomething.org:8888/api/v1/reportback-items/6789",
                            "media": {
                                "uri": "http://dev.dosomething.org:8888/sites/default/files/styles/480x480/public/reportbacks/1631/uid_1706281-nid_1631-run_nid_1903-1.jpg?itok=Kg0x4_De",
                                "type": "image"
                            },
                            "created_at": "1458672694",
                            "kudos": {
                                "data": {
                                    "641": {
                                        "term": {
                                            "id": "641",
                                            "name": "heart",
                                            "total": 2
                                        },
                                        "current_user": null
                                    }
                                }
                            }
                        },
                        {
                            "id": "6793",
                            "status": "approved",
                            "caption": "bunny",
                            "uri": "http://dev.dosomething.org:8888/api/v1/reportback-items/6793",
                            "media": {
                                "uri": "http://dev.dosomething.org:8888/sites/default/files/styles/480x480/public/reportbacks/1631/uid_1706281-nid_1631-run_nid_1903-2_2.jpg?itok=mTePvZi5",
                                "type": "image"
                            },
                            "created_at": "1458672945",
                            "kudos": {
                                "data": {
                                    "641": {
                                        "term": {
                                            "id": "641",
                                            "name": "heart",
                                            "total": 2
                                        },
                                        "current_user": null
                                    }
                                }
                            }
                        },
                        {
                            "id": "6795",
                            "status": "excluded",
                            "caption": "BFG",
                            "uri": "http://dev.dosomething.org:8888/api/v1/reportback-items/6795",
                            "media": {
                                "uri": "http://dev.dosomething.org:8888/sites/default/files/styles/480x480/public/reportbacks/1631/uid_1706281-nid_1631-run_nid_1903-3_0.jpg?itok=Rm4VQuMu",
                                "type": "image"
                            },
                            "created_at": "1458673023",
                            "kudos": {
                                "data": {
                                    "641": {
                                        "term": {
                                            "id": "641",
                                            "name": "heart",
                                            "total": 2
                                        },
                                        "current_user": null
                                    }
                                }
                            }
                        },
                        {
                            "id": "6796",
                            "status": "approved",
                            "caption": "duuuude",
                            "uri": "http://dev.dosomething.org:8888/api/v1/reportback-items/6796",
                            "media": {
                                "uri": "http://dev.dosomething.org:8888/sites/default/files/styles/480x480/public/reportbacks/1631/uid_1706281-nid_1631-run_nid_1903-4.jpg?itok=Pn4w_pR3",
                                "type": "image"
                            },
                            "created_at": "1458673094",
                            "kudos": {
                                "data": {
                                    "641": {
                                        "term": {
                                            "id": "641",
                                            "name": "heart",
                                            "total": 2
                                        },
                                        "current_user": null
                                    }
                                }
                            }
                        },
                        {
                            "id": "6797",
                            "status": "approved",
                            "caption": "silly wabbit",
                            "uri": "http://dev.dosomething.org:8888/api/v1/reportback-items/6797",
                            "media": {
                                "uri": "http://dev.dosomething.org:8888/sites/default/files/styles/480x480/public/reportbacks/1631/uid_1706281-nid_1631-run_nid_1903-5.jpg?itok=v097VpDc",
                                "type": "image"
                            },
                            "created_at": "1458686313",
                            "kudos": {
                                "data": {
                                    "641": {
                                        "term": {
                                            "id": "641",
                                            "name": "heart",
                                            "total": 2
                                        },
                                        "current_user": null
                                    }
                                }
                            }
                        },
                        {
                            "id": "6798",
                            "status": "approved",
                            "caption": "I'm so small!",
                            "uri": "http://dev.dosomething.org:8888/api/v1/reportback-items/6798",
                            "media": {
                                "uri": "http://dev.dosomething.org:8888/sites/default/files/styles/480x480/public/reportbacks/1631/uid_1706281-nid_1631-run_nid_1903-6.jpg?itok=ClZ18h3D",
                                "type": "image"
                            },
                            "created_at": "1458686433",
                            "kudos": {
                                "data": {
                                    "641": {
                                        "term": {
                                            "id": "641",
                                            "name": "heart",
                                            "total": 2
                                        },
                                        "current_user": null
                                    }
                                }
                            }
                        }
                    ]
                }
            },
            "user": {
                "id": "56f18b9da59dbf59048b456c",
                "drupal_id": "1706281",
                "first_name": "Roberto",
                "last_initial": "G",
                "photo": null,
                "country": "US"
            }
        }
    ]
}
```

#### How should this be reviewed?
Make sure everything is safely behind the Rogue feature flag. Please compare the responses and make sure the format is identical. Make sure everything behaves properly if Rogue throws an error. 

#### Relevant tickets
[Trello Card](https://trello.com/c/0Nc6Xjdg/484-as-a-developer-i-want-to-make-sure-phoenixs-signups-endpoint-has-a-rogue-proxy-for-gladiator-to-still-work-when-rogue-is-turned)

#### Checklist
- [ ] Documentation added for new features/changed endpoints.
- [ ] Tested on staging.
- [ ] Pinged a PM if this is a larger PR that would benefit from some additional testing love
- [ ] Post a message in #phoenix if this includes something that causes a rebuild!

### CODE REVIEW

File: lib/modules/dosomething/dosomething_rogue/dosomething_rogue.module [hunk 1]
```diff
@@ -60,6 +61,205 @@ function dosomething_rogue_get_reportback_items($inputs)
...
+    'page' => dosomething_helpers_isset($inputs, 'page'),
+  ];
+
+  $response = $client->getActivity($rogue_inputs);
+
+  return $response;
+}
+
+/**
+  * Send a GET request to return all activiy matching a given
```
Comment (sbsmith86) [inline lib/modules/dosomething/dosomething_rogue/dosomething_rogue.module]: Is this actually sending a request or just formatting the response?

Comment (katiecrane) [reply lib/modules/dosomething/dosomething_rogue/dosomething_rogue.module]: @sbsmith86 good catch! this was accurate at some point but not anymore

File: lib/modules/dosomething/dosomething_rogue/dosomething_rogue.module [hunk 2]
```diff
@@ -60,6 +61,205 @@ function dosomething_rogue_get_reportback_items($inputs)
...
+              'id' => $post['id'],
+              'status' => dosomething_rogue_transform_rogue_status_to_phoenix_status($post['status']),
+              'caption' => $post['media']['caption'],
+              'uri' => null,
+              'media' => [
+                'uri' => $post['media']['url'],
+                'type' => 'image',
+              ],
+              'created_at' => strtotime($post['created_at']),
+              // @TODO: include kudo stuff here CONDITIONALLY
```
Comment (sbsmith86) [inline lib/modules/dosomething/dosomething_rogue/dosomething_rogue.module]: Should we make a separate trello card for this?

File: lib/modules/dosomething/dosomething_rogue/dosomething_rogue.module [hunk 3]
```diff
@@ -60,6 +61,205 @@ function dosomething_rogue_get_reportback_items($inputs)
...
+            // Include reaction data if there is any
+            if (count($post['reactions']) > 0) {
+              $post_data['kudos']['data'] = [
+                '641' => [
+                  'term' => [
+                    'id' => '641',
+                    'name' => 'heart',
+                    'total' => count($post['reactions']),
+                  ],
+                  'current_user' => null, // @TODO: there's not really a current user in this case so always null is cool yes?
```
Comment (sbsmith86) [inline lib/modules/dosomething/dosomething_rogue/dosomething_rogue.module]: Yes, I think this is fine.

File: lib/modules/dosomething/dosomething_rogue/dosomething_rogue.module (hunk 2 above)
Comment (katiecrane) [reply lib/modules/dosomething/dosomething_rogue/dosomething_rogue.module]: @sbsmith86 ah another comment that needs to go! we actually _do_ want to include an empty array like this if there are no reactions, and we add any actual data in below

File: lib/modules/dosomething/dosomething_rogue/dosomething_rogue.module [hunk 4]
```diff
@@ -60,6 +61,202 @@ function dosomething_rogue_get_reportback_items($inputs)
...
+      // Get campaign and run information for the signup data response
+        $northstar_user = dosomething_northstar_get_user($signup['northstar_id']);
+
+        // Get campaign information
+        $drupal_user = dosomething_user_get_user_by_northstar_id($signup['northstar_id']);
+
+        try {
+          $campaign_run = dosomething_helpers_get_current_campaign_run_for_user($signup['campaign_id'], $drupal_user); // @TODO - get errors when calling this sometimes it seems (500 - no campaign data found??????)
+        } catch(Exception $e) {
+          // the regular Phoenix endpoint doesn't seem to handle this so should we just leave it? unclear
```
Comment (sbsmith86) [inline lib/modules/dosomething/dosomething_rogue/dosomething_rogue.module]: Yea, I'm pretty sure the phoenix endpoint doesn't support runs. let's leave it unhandled.

**Review by sbsmith86 — APPROVED** [review]
This looks awesome to me! Great work.

File: lib/modules/dosomething/dosomething_rogue/dosomething_rogue.module (hunk 4 above)
Comment (katiecrane) [reply lib/modules/dosomething/dosomething_rogue/dosomething_rogue.module]: @sbsmith86 do you mean to leave it like this or to remove the try/catch here?

Comment (sbsmith86) [reply lib/modules/dosomething/dosomething_rogue/dosomething_rogue.module]: I think you can leave it like it is.

Comment (katiecrane) [reply lib/modules/dosomething/dosomething_rogue/dosomething_rogue.module]: @sbsmith86 cool! just updated the comments

**Comment by katiecrane** [comment]
@sbsmith86 I've only updated comments/docs since you approved, does the approval still stand?

**Comment by sbsmith86** [comment]
Yerp!

**Pull request merged**